examples/kitchen-sink.yaml

```diff
@@ -426,3 +426,37 @@ resource:
         - process.command_args
   # Configure the resource schema URL.
   schema_url: https://opentelemetry.io/schemas/1.16.0
+# Configure instrumentation
+instrumentation:
+  # Configure options common to multiple languages
+  common:
+    peer-service-mapping:
```

Comment:
What do you think about organizing these by top level semconv namespace? So something like:
```
instrumentation:
  general:
    peer:
      service-mapping:
        - peer: foo
           service: bar
    http:
      client:
        captured-request-headers: 
          - X-Foo
        captured-response-headers: ... // omitted for brevity
      server:
        captured-request-headers: ... // omitted for brevity
        captured-response-headers: ... // omitted for brevity
  java: ..
  php: ..
```
Within a language, it makes sense to organize by instrumentation library like you have. This produces asymmetry, with the children of `instrumentation.general` being semconv namespaces, and `instrumentation.[language]` being instrumentation library names. One alternative could be to get rid of the language specific sections, and instead jump directly to configuration by instrumentation library names. With this, we could have a "semconv" configuration section, and instrumentation library configuration section:
```
instrumentation:
  semconv:
    peer:
      service-mapping:
        - peer: foo
           service: bar
    http:
      client:
        captured-request-headers: 
          - X-Foo
        captured-response-headers: ... // omitted for brevity
      server:
        captured-request-headers: ... // omitted for brevity
        captured-response-headers: ... // omitted for brevity
  libraries:
    logback-appender: ..
    example_instrumentation: ..
```
This could be problematic if the same instrumentation library key name is used in different languages but with different configuration schemas.

Reply:
Organising general config to match semconv namespaces is a good idea, I've made that change.
My main concern with everything else being a free-for-all is conflict between keys across different languages, which we're unlikely to find out about until a user trips up on it.
A possible benefit of having a reserved space per language is that each language should be able to safely only process `general` plus their own language section for instrumentation config...?

Reply:
> A possible benefit of having a reserved space per language is that each language should be able to safely only process general plus their own language section for instrumentation config...?

Yes that's true. Also, it makes explicit to the configuration author that they're configuring something language specific. Sure it results in an additional layer of nesting, but I think the positives outweigh this.